app/mailboxes/reply_mailbox.rb

```diff
@@ -26,6 +26,8 @@ def find_relative_conversation
   end
 
   def conversation_uuid_from_to_address
+    return if mail.to.nil?
```

Comment:
if `mail.to.nil?` we might need to look into the `X-forwarded-for` header I believe. Lets add a method to parse the `to` address in our mail decorator as well. 

cc: @pranavrajs

Reply:
Yes, please check if mail gem already handles this, otherwise let us add a fallback.

Reply:
Okay, let me make this change and I'll add the specs for this new decorator.